src/mesh/voxel_cloud.rs

```diff
@@ -758,303 +491,316 @@ impl VoxelCloud {
+    let x_dimension_i32 = cast_i32(block_dimensions.x);
+    let z = one_dimensional_i32 / horizontal_area_i32;
+    let y = (one_dimensional_i32 % horizontal_area_i32) / x_dimension_i32;
+    let x = one_dimensional_i32 % x_dimension_i32;
+    Point3::new(x, y, z)
 }
 
-/// Gets the index to the voxel map from absolute voxel coordinates
-/// (relative to the voxel space origin).
+/// Computes a voxel position relative to the model space origin (absolute
+/// coordinate) from an index to the linear representation of the voxel block.
 ///
 /// Returns None if out of bounds.
-fn absolute_three_dimensional_coordinate_to_one_dimensional(
-    absolute_coords: &Point3<i32>,
+fn one_dimensional_to_absolute_voxel_coordinate(
+    one_dimensional_coordinate: usize,
     block_start: &Point3<i32>,
     block_dimensions: &Vector3<u32>,
-) -> Option<usize> {
-    let relative_coords = absolute_coords - block_start.coords;
-    relative_three_dimensional_coordinate_to_one_dimensional(&relative_coords, block_dimensions)
+) -> Point3<i32> {
+    let relative =
+        one_dimensional_to_relative_voxel_coordinate(one_dimensional_coordinate, block_dimensions);
+    relative_voxel_to_absolute_voxel_coordinate(&relative, block_start)
 }
 
-/// Computes a voxel position relative to the model space origin (absolute
-/// coordinate) from an index to the linear representation of the voxel
-/// block.
+/// Computes a voxel position in world space cartesian units from an index to
+/// the linear representation of the voxel block.
 ///
 /// Returns None if out of bounds.
-fn one_dimensional_to_absolute_three_dimensional_coordinate(
-    one_dimensional: usize,
+fn one_dimensional_to_cartesian_coordinate(
+    one_dimensional_coordinate: usize,
     block_start: &Point3<i32>,
     block_dimensions: &Vector3<u32>,
-) -> Option<Point3<i32>> {
-    one_dimensional_to_relative_three_dimensional_coordinate(one_dimensional, block_dimensions)
-        .map(|relative| relative + block_start.coords)
+    voxel_dimensions: &Vector3<f32>,
+) -> Point3<f32> {
+    let relative =
+        one_dimensional_to_relative_voxel_coordinate(one_dimensional_coordinate, block_dimensions);
+    relative_voxel_to_cartesian_coordinate(&relative, block_start, voxel_dimensions)
 }
 
-/// Computes the voxel-space coordinates of a voxel containing the input
-/// point.
-fn cartesian_to_absolute_voxel_coords(
-    point: &Point3<f32>,
-    voxel_dimensions: &Vector3<f32>,
+/// Computes the center of a voxel in absolute voxel units from voxel
+/// coordinates relative to the voxel block start.
+fn relative_voxel_to_absolute_voxel_coordinate(
+    relative_coordinate: &Point3<i32>,
+    block_start: &Point3<i32>,
 ) -> Point3<i32> {
+    relative_coordinate + block_start.coords
+}
+
+/// Computes the center of a voxel in worlds space cartesian units from voxel
+/// coordinates relative to the voxel block start.
+///
+/// # Panics
+/// Panics if any of the voxel dimensions is zero.
```

Comment:
The next function requires voxel dimension components to not be equal to or bewlow zero, which seems more correct. Let's update this one,